src/components/swap/SwapLineItem.tsx

```diff
@@ -0,0 +1,250 @@
+import { NumberType, useFormatter } from 'utils/formatNumbers'
+import { getPriceImpactColor } from 'utils/prices'
+
+import { GasBreakdownTooltip, UniswapXDescription } from './GasBreakdownTooltip'
+import SwapRoute from './SwapRoute'
+
+export enum SwapLineItemTypes {
+  EXCHANGE_RATE,
+  NETWORK_FEE,
+  INPUT_TOKEN_FEE_ON_TRANSFER,
+  OUTPUT_TOKEN_FEE_ON_TRANSFER,
+  PRICE_IMPACT,
+  MAXIMUM_INPUT,
+  MINIMUM_OUTPUT,
+  EXPECTED_OUTPUT,
+  ROUTING_INFO,
+}
+
+const DetailRowValue = styled(ThemedText.BodySmall)`
+  text-align: right;
+  overflow-wrap: break-word;
+`
+const LabelText = styled(ThemedText.BodySmall)<{ hasTooltip?: boolean }>`
+  cursor: ${({ hasTooltip }) => (hasTooltip ? 'help' : 'auto')};
+  color: ${({ theme }) => theme.neutral2};
+`
+const ColorWrapper = styled.span<{ textColor?: keyof DefaultTheme }>`
+  ${({ textColor, theme }) => textColor && `color: ${theme[textColor]};`}
+`
+
+function FOTTooltipContent() {
+  return (
+    <>
+      <Trans>
+        Some tokens take a fee when they are bought or sold, which is set by the token issuer. Uniswap does not receive
+        any of these fees.
+      </Trans>{' '}
+      <ExternalLink href="https://support.uniswap.org/hc/en-us/articles/18673568523789-What-is-a-token-fee-">
+        Learn more
+      </ExternalLink>
+    </>
+  )
+}
+
+function Loading({ width = 50 }: { width?: number }) {
+  return <LoadingRow data-testid="loading-row" height={15} width={width} />
+}
+
+function ExchangeRateRow({ trade }: { trade: InterfaceTrade }) {
+  const { formatNumber } = useFormatter()
+  const rate = `1 ${trade.executionPrice.quoteCurrency.symbol} = ${formatNumber({
+    input: parseFloat(trade.executionPrice.toFixed(9)),
+    type: NumberType.TokenTx,
+  })} ${trade.executionPrice.baseCurrency.symbol}`
+  return <>{rate}</>
+}
+
+function ColoredPercentRow({ percent }: { percent: Percent }) {
+  const { formatPriceImpact } = useFormatter()
+  return <ColorWrapper textColor={getPriceImpactColor(percent)}>{formatPriceImpact(percent)}</ColorWrapper>
+}
+
+function CurrencyAmountRow({ amount }: { amount: CurrencyAmount<Currency> }) {
+  const { formatCurrencyAmount } = useFormatter()
+  const formattedAmount = formatCurrencyAmount({ amount, type: NumberType.SwapDetailsAmount })
+  return <>{`${formattedAmount} ${amount.currency.symbol}`}</>
+}
+
+type LineItemData = {
+  Label: React.FC
+  Value: React.FC
+  TooltipBody?: React.FC
+  tooltipSize?: TooltipSize
+  loaderWidth?: number
+}
+
+function useLineItem(props: SwapLineItemProps): LineItemData | undefined {
+  const { trade, syncing, allowedSlippage, type } = props
+  const { formatNumber } = useFormatter()
+
+  const isUniswapX = isUniswapXTrade(trade)
+  const isPreview = isPreviewTrade(trade)
+  const chainId = trade.inputAmount.currency.chainId
+
+  // Used to hide price impact row if last trade was UniswapX
+  const [wasUniswapX, setWasUniswapX] = useState<boolean>(isUniswapX)
```

Comment:
maybe relabel this as `hasPriceImpact` because preview trades also dont have price impact

Reply:
_now_ its called `lastSubmittableFillType` because it was still confusing